src/api/handlers/misc.py

```diff
@@ -157,3 +161,50 @@ def authenticateFrontendInTestMode(self, request):
             "token", value=token, max_age=24 * 60 * 60, samesite="Strict"
         )
         return response
+    
+    
+    def load_essential_initial_site_data(self, request):
+        context: Context = request.context
+        args: dict = context.args
+        
+        self.validator.expect("community_id", is_required=False)
+        self.validator.expect("subdomain", is_required=False)
+        self.validator.expect("data", 'str_list', is_required=True)
+        self.validator.expect("id", is_required=False)
+        
+        args, err = self.validator.verify(args, strict=True)
+        if err:
+            return err
+        
+        endpoints = args.pop("data", [])
+    
+        def fetch_data(endpoint):
+            endpoint = request.build_absolute_uri(endpoint)
+            response = requests.post(endpoint,data=args)
+            return response.json()
```

Comment:
if we do this, we lose the cookies of the user right?  can we make sure the cookies of the user are passed through?  so in essence, we assume their role and make those requests on their behalf?

Reply:
I'll conduct tests with authenticated users to verify whether the cookie is being passed. If it's not, I'll ensure that it's included in the request